LISAv2 abnormal exit due to unhandled exception

```
05/21/2019 12:11:00 : [INFO ] Calling function - Upload-RemoteFile. Error in upload after 10 attempt, hence giving up

05/21/2019 12:11:00 : [INFO ] dos2unix: converting file .\Testscripts\Linux\CollectLogFile.sh to Unix format...

05/21/2019 12:11:00 : [INFO ] Uploading .\Testscripts\Linux\CollectLogFile.sh to lisa : 52.191.136.173, port 22 using Password authentication

05/21/2019 12:11:02 : [WARN ] Error in upload, attempt 1, retrying

05/21/2019 12:11:12 : [INFO ] Uploading .\Testscripts\Linux\CollectLogFile.sh to lisa : 52.191.136.173, port 22 using Password authentication

05/21/2019 12:11:15 : [WARN ] Error in upload, attempt 2, retrying

05/21/2019 12:11:25 : [INFO ] Uploading .\Testscripts\Linux\CollectLogFile.sh to lisa : 52.191.136.173, port 22 using Password authentication

05/21/2019 12:11:27 : [WARN ] Error in upload, attempt 3, retrying

05/21/2019 12:11:37 : [INFO ] Uploading .\Testscripts\Linux\CollectLogFile.sh to lisa : 52.191.136.173, port 22 using Password authentication

05/21/2019 12:11:39 : [WARN ] Error in upload, attempt 4, retrying

05/21/2019 12:11:49 : [INFO ] Uploading .\Testscripts\Linux\CollectLogFile.sh to lisa : 52.191.136.173, port 22 using Password authentication

05/21/2019 12:11:52 : [ERROR] EXCEPTION: Calling function - Upload-RemoteFile. Error in upload after 5 attempt, hence giving up

05/21/2019 12:11:52 : [ERROR] Source: Line 125 in script .\Libraries\TestHelpers.psm1.

05/21/2019 12:11:52 : [INFO ] Copying all files back to original working directory: C:\Jenkins-Slaves\azure-slave-2-executor-7\workspace\tool-run-lisav2-tests-in-parallel@27.

05/21/2019 12:12:22 : [INFO ] Setting workspace back to original location: C:\Jenkins-Slaves\azure-slave-2-executor-7\workspace\tool-run-lisav2-tests-in-parallel@27

05/21/2019 12:12:22 : [INFO ] Forcefully exiting with exit code 0 as ExitWithZero flag was set to True

05/21/2019 12:12:22 : [INFO ] LISAv2 exit code: 0
```

--- a/TestControllers/TestController.psm1
+++ b/TestControllers/TestController.psm1
@@ -632,70 +632,80 @@ Class TestController
 	}
 
 	[void] GetSystemBasicLogs($AllVMData, $User, $Password, $CurrentTestData, $CurrentTestResult, $enableTelemetry) {
-        if ($allVMData.Count -gt 1) {
-			$vmData = $allVMData[0]
-		} else {
-			$vmData = $allVMData
-		}
-		$FilesToDownload = "$($vmData.RoleName)-*.txt"
-		Copy-RemoteFiles -upload -uploadTo $vmData.PublicIP -port $vmData.SSHPort `
-			-files .\Testscripts\Linux\CollectLogFile.sh `
-			-username $user -password $password -maxRetry 5 | Out-Null
-		$Null = Run-LinuxCmd -username $user -password $password -ip $vmData.PublicIP -port $vmData.SSHPort `
-			-command "bash CollectLogFile.sh -hostname $($vmData.RoleName)" -ignoreLinuxExitCode -runAsSudo
-		$Null = Copy-RemoteFiles -downloadFrom $vmData.PublicIP -port $vmData.SSHPort `
-			-username $user -password $password -files "$FilesToDownload" -downloadTo $global:LogDir -download
-		$KernelVersion = Get-Content "$global:LogDir\$($vmData.RoleName)-kernelVersion.txt"
-		$HardwarePlatform = Get-Content "$global:LogDir\$($vmData.RoleName)-hardwarePlatform.txt"
-		$GuestDistro = Get-Content "$global:LogDir\$($vmData.RoleName)-distroVersion.txt"
-		$LISMatch = (Select-String -Path "$global:LogDir\$($vmData.RoleName)-lis.txt" -Pattern "^version:").Line
-		if ($LISMatch) {
-			$LISVersion = $LISMatch.Split(":").Trim()[1]
-		} else {
-			$LISVersion = "NA"
-		}
-		#region Host Version checking
-		$HostVersion = ""
-		$FoundLineNumber = (Select-String -Path "$global:LogDir\$($vmData.RoleName)-dmesg.txt" -Pattern "Hyper-V Host Build").LineNumber
-		if (![string]::IsNullOrEmpty($FoundLineNumber)) {
-			$ActualLineNumber = $FoundLineNumber[-1] - 1
-			$FinalLine = [string]((Get-Content -Path "$global:LogDir\$($vmData.RoleName)-dmesg.txt")[$ActualLineNumber])
-			$FinalLine = $FinalLine.Replace('; Vmbus version:4.0','')
-			$FinalLine = $FinalLine.Replace('; Vmbus version:3.0','')
-			$HostVersion = ($FinalLine.Split(":")[$FinalLine.Split(":").Count -1 ]).Trim().TrimEnd(";")
-		}
-		#endregion
-
-		if ($currentTestData.AdditionalHWConfig.Networking -imatch "SRIOV") {
-			$Networking = "SRIOV"
-		} else {
-			$Networking = "Synthetic"
-		}
-		$VMSize = ""
-		if ($global:TestPlatform -eq "Azure") {
-			$VMSize = $vmData.InstanceSize
-		}
-		if ($global:TestPlatform -eq "HyperV") {
-			$VMSize = $global:HyperVInstanceSize
-		}
-		$VMGeneration = $vmData.VMGeneration
-		#endregion
-		if ($enableTelemetry) {
-			$dataTableName = ""
-			if ($this.XmlSecrets.secrets.TableName) {
-				$dataTableName = $this.XmlSecrets.secrets.TableName
-				Write-LogInfo "Using table name from secrets: $dataTableName"
+		try {
+			if ($allVMData.Count -gt 1) {
+				$vmData = $allVMData[0]
 			} else {
-				$dataTableName = "LISAv2Results"
+				$vmData = $allVMData
 			}
+			$FilesToDownload = "$($vmData.RoleName)-*.txt"
+			Copy-RemoteFiles -upload -uploadTo $vmData.PublicIP -port $vmData.SSHPort `
+				-files .\Testscripts\Linux\CollectLogFile.sh `
+				-username $user -password $password -maxRetry 5 | Out-Null
+			$Null = Run-LinuxCmd -username $user -password $password -ip $vmData.PublicIP -port $vmData.SSHPort `
+				-command "bash CollectLogFile.sh -hostname $($vmData.RoleName)" -ignoreLinuxExitCode -runAsSudo
+			$Null = Copy-RemoteFiles -downloadFrom $vmData.PublicIP -port $vmData.SSHPort `
+				-username $user -password $password -files "$FilesToDownload" -downloadTo $global:LogDir -download
+			$KernelVersion = Get-Content "$global:LogDir\$($vmData.RoleName)-kernelVersion.txt"
+			$HardwarePlatform = Get-Content "$global:LogDir\$($vmData.RoleName)-hardwarePlatform.txt"
+			$GuestDistro = Get-Content "$global:LogDir\$($vmData.RoleName)-distroVersion.txt"
+			$LISMatch = (Select-String -Path "$global:LogDir\$($vmData.RoleName)-lis.txt" -Pattern "^version:").Line
+			if ($LISMatch) {
+				$LISVersion = $LISMatch.Split(":").Trim()[1]
+			} else {
+				$LISVersion = "NA"
+			}
+			#region Host Version checking
+			$HostVersion = ""
+			$FoundLineNumber = (Select-String -Path "$global:LogDir\$($vmData.RoleName)-dmesg.txt" -Pattern "Hyper-V Host Build").LineNumber
+			if (![string]::IsNullOrEmpty($FoundLineNumber)) {
+				$ActualLineNumber = $FoundLineNumber[-1] - 1
+				$FinalLine = [string]((Get-Content -Path "$global:LogDir\$($vmData.RoleName)-dmesg.txt")[$ActualLineNumber])
+				$FinalLine = $FinalLine.Replace('; Vmbus version:4.0', '')
+				$FinalLine = $FinalLine.Replace('; Vmbus version:3.0', '')
+				$HostVersion = ($FinalLine.Split(":")[$FinalLine.Split(":").Count - 1 ]).Trim().TrimEnd(";")
+			}
+			#endregion
 
-			$SQLQuery = Get-SQLQueryOfTelemetryData -TestPlatform $global:TestPlatform -TestLocation $global:TestLocation -TestCategory $CurrentTestData.Category `
-                -TestArea $CurrentTestData.Area -TestName $CurrentTestData.TestName -CurrentTestResult $CurrentTestResult `
-                -ExecutionTag $global:GlobalConfig.Global.$global:TestPlatform.ResultsDatabase.testTag -GuestDistro $GuestDistro -KernelVersion $KernelVersion `
-                -HardwarePlatform $HardwarePlatform -LISVersion $LISVersion -HostVersion $HostVersion -VMSize $VMSize -VMGeneration $VMGeneration -Networking $Networking `
-                -ARMImageName $global:ARMImageName -OsVHD $global:BaseOsVHD -BuildURL $env:BUILD_URL -TableName $dataTableName
+			if ($currentTestData.AdditionalHWConfig.Networking -imatch "SRIOV") {
+				$Networking = "SRIOV"
+			} else {
+				$Networking = "Synthetic"
+			}
+			$VMSize = ""
+			if ($global:TestPlatform -eq "Azure") {
+				$VMSize = $vmData.InstanceSize
+			}
+			if ($global:TestPlatform -eq "HyperV") {
+				$VMSize = $global:HyperVInstanceSize
+			}
+			$VMGeneration = $vmData.VMGeneration
+			#endregion
+			if ($enableTelemetry) {
+				$dataTableName = ""
+				if ($this.XmlSecrets.secrets.TableName) {
+					$dataTableName = $this.XmlSecrets.secrets.TableName
+					Write-LogInfo "Using table name from secrets: $dataTableName"
+				} else {
+					$dataTableName = "LISAv2Results"
+				}
 
-			Upload-TestResultToDatabase -SQLQuery $SQLQuery
+				$SQLQuery = Get-SQLQueryOfTelemetryData -TestPlatform $global:TestPlatform -TestLocation $global:TestLocation -TestCategory $CurrentTestData.Category `
+					-TestArea $CurrentTestData.Area -TestName $CurrentTestData.TestName -CurrentTestResult $CurrentTestResult `
+					-ExecutionTag $global:GlobalConfig.Global.$global:TestPlatform.ResultsDatabase.testTag -GuestDistro $GuestDistro -KernelVersion $KernelVersion `
+					-HardwarePlatform $HardwarePlatform -LISVersion $LISVersion -HostVersion $HostVersion -VMSize $VMSize -VMGeneration $VMGeneration -Networking $Networking `
+					-ARMImageName $global:ARMImageName -OsVHD $global:BaseOsVHD -BuildURL $env:BUILD_URL -TableName $dataTableName
+
+				Upload-TestResultToDatabase -SQLQuery $SQLQuery
+			}
 		}
-    }
+		catch {
+			$line = $_.InvocationInfo.ScriptLineNumber
+			$script_name = ($_.InvocationInfo.ScriptName).Replace($PWD,".")
+			$ErrorMessage =  $_.Exception.Message
+			Write-LogErr "EXCEPTION: $ErrorMessage"
+			Write-LogErr "Calling function - $($MyInvocation.MyCommand)."
+			Write-LogErr "Source: Line $line in script $script_name."
+		}
+  }
 }

